reconcile/utils/jinja2/utils.py

```diff
@@ -188,6 +193,118 @@ def list_s3_objects(
+    with open(output_file_path, encoding="utf-8") as f:
+        content = f.read()
+        # Strip leading whitespace and document separator since we're embedding in existing YAML
+        content = content.lstrip()
+        content = content.removeprefix("---\n")
+        # Remove sloth_* labels and title annotation
+        content = re.sub(r"^(\s+)sloth_\w+:.*\n", "", content, flags=re.MULTILINE)
+        content = re.sub(
+            r"^(\s+)title:.*\n(?:(?:\1\s+.*\n)*)", "", content, flags=re.MULTILINE
+        )
+        # Remove labels sections from recording rules
+        # sloth adds numerous labels that violate schema and we don't have any functionality based off recording rule labels
+        content = re.sub(
+            r"^(\s+)labels:\s*\n(?=\s*(?:- (?:record|alert|name):|$))",
+            "",
+            content,
+            flags=re.MULTILINE,
+        )
+        return content
+
+
+def sloth_alerts(
+    service: str,
+    slo_name: str,
+    objective: float,
+    error_query: str,
+    total_query: str,
+    version: str = "prometheus/v1",
+) -> str:
+    """Generate Prometheus rules using sloth: https://sloth.dev
+
+    Args:
+        service: Service name identifier
+        slo_name: Name of the SLO
+        objective: Target percentage (e.g. 99.9)
+        error_query: Prometheus query for error events
+        total_query: Prometheus query for total events
+        version: Spec version (default: "prometheus/v1")
+
+    Returns:
+        Generated Prometheus rules as YAML string
+    """
+    # Build the SLO definition
+    slo = {
+        "name": slo_name,
+        "objective": objective,
+        "description": f"{slo_name} SLO for {service}",
+        "sli": {
+            "events": {
+                "error_query": error_query.replace("{{window}}", "{{.window}}"),
+                "total_query": total_query.replace("{{window}}", "{{.window}}"),
+            }
+        },
+        "alerting": {
+            "name": f"{service.title()}{slo_name.title()}",
+            "annotations": {
+                "summary": f"High error rate on '{service}' {slo_name}",
+                "message": f"High error rate on '{service}' {slo_name}",
+            },
+            "page_alert": {
+                "labels": {
+                    "severity": "critical",
+                    "service": service,
+                    "slo": slo_name,
+                }
+            },
+            "ticket_alert": {
+                "labels": {
+                    "severity": "medium",
+                    "service": service,
+                    "slo": slo_name,
+                }
+            },
+        },
+    }
+
+    spec = {
+        "version": version,
+        "service": service,
+        "slos": [slo],
+    }
+
+    # Generate rules using temporary files
+    input_fd, input_path = tempfile.mkstemp(suffix=".yml")
+    output_fd, output_path = tempfile.mkstemp(suffix=".yml")
```

Comment:
use `NamedTemporaryFile` and context manager `with` instead of `mkstemp` to let it auto cleanup.